src/main/java/subway/domain/section/add/strategy/MidAddSectionStrategy.java

```diff
@@ -0,0 +1,18 @@
+package subway.domain.section.add.strategy;
+
+import subway.domain.section.Section;
+import subway.domain.section.Sections;
+
+public class MidAddSectionStrategy implements AddSectionStrategy {
+
+    @Override
+    public void addSection(final Sections sections, final Section insertSection) {
+        final Section originSection = sections.findOriginSection(insertSection);
+        final Section upSection = insertSection.createUpSection(originSection);
+        final Section downSection = insertSection.createDownSection(originSection);
+        final int originIndex = sections.findIndex(originSection);
+        sections.delete(originSection);
+        sections.addSection(originIndex, upSection);
+        sections.addSection(originIndex + 1, downSection);
```

Comment:
이부분도 유사한 역할을 하는 부분끼리 묶어 메서드로 한번 더 분리해봐도 좋을 것 같아요! 흐름이 읽히도록요

Reply:
이 부분은 메서드로 분리하면 파라미터가 너무 많이 생겨서 어떻게 분리해야할지 모르겠네요..
`createUpSection`이랑 `createDownSection`을 `createSection`으로 바꾸는 것도 생각해봤는데, 여기서도 파라미터가 많이 생겨서 줄바꿈으로 구분했습니다!

```suggestion
        final Section originSection = sections.findOriginSection(insertSection);
        final int originIndex = sections.findIndex(originSection);
        sections.delete(originSection);

        final Section upSection = insertSection.createUpSection(originSection);
        sections.addSection(originIndex, upSection);

        final Section downSection = insertSection.createDownSection(originSection);
        sections.addSection(originIndex + 1, downSection);
```